server/routers/activity.ts

```diff
@@ -39,6 +61,7 @@ export const activityRouter = createTRPCRouter({
       const updatedActivity = await ctx.prisma.activity.update({
         where: {
           id: input.id,
+          userId: ctx.user?.id,
```

Comment:
Same as above. "Because it would mean user is trying to update activity which is NOT his"